compiler/rustc_metadata/src/creader.rs

```diff
@@ -60,16 +60,22 @@ impl std::fmt::Debug for CStore {
     }
 }
 
-pub struct CrateLoader<'a> {
+pub struct CrateLoader<'a, 'tcx: 'a> {
     // Immutable configuration.
-    sess: &'a Session,
-    metadata_loader: &'a MetadataLoaderDyn,
-    local_crate_name: Symbol,
+    tcx: TyCtxt<'tcx>,
     // Mutable output.
     cstore: &'a mut CStore,
     used_extern_options: &'a mut FxHashSet<Symbol>,
 }
 
+impl<'a, 'tcx> std::ops::Deref for CrateLoader<'a, 'tcx> {
```

Comment:
Any strong reasons to add this method?
It doesn't look like `CrateLoader` is used as `TyCtxt` often.

Reply:
Ah, this is just to avoid replacing `self.sess` with `self.tcx.sess` everywhere.

I can do a PR hiding all session accesses behind a method before this PR or I can remove the Deref impl in a follow up